test/GoodGhosting_Funds_Redeemed_From_Contract_On_EarlyWithdraw.test.js

```diff
@@ -100,20 +101,46 @@ contract("GoodGhosting", (accounts) => {
                         "loser unable to early withdraw from game",
                     );
                 }
+
                 // since funds haven't been deposited to external pool yet so the funds will be redeemed from the contract directly
                 await goodGhosting.depositIntoExternalPool({ from: admin });
 
                 // j must start at 1 - Player1 (index 0) early withdraw, so won't continue making deposits
-                for (let j = 1; j < players.length; j++) {
-                    const player = players[j];
-                    const depositResult = await goodGhosting.makeDeposit({ from: player });
-                    truffleAssert.eventEmitted(
-                        depositResult,
-                        "Deposit",
-                        (ev) => ev.player === player && ev.segment.toNumber() === segmentIndex,
-                        `player ${j} unable to deposit for segment ${segmentIndex}`,
-                    );
+                if (segmentIndex === 1) {
+                    // All players pay for the 1st segment
+                    for (let j = 1; j < players.length; j++) {
+                        const player = players[j];
+                        const depositResult = await goodGhosting.makeDeposit({ from: player });
+                        truffleAssert.eventEmitted(
+                            depositResult,
+                            "Deposit",
+                            (ev) => ev.player === player && ev.segment.toNumber() === segmentIndex,
+                            `player ${j} unable to deposit for segment ${segmentIndex}`,
+                        );
+                    }
+                } else if (segmentIndex > 1) {
+                    // Segment 2 onwards secondLoser withdraws from the game and since segment deposit is < than withdrawn amount the segment deposit is set to 0
+                    if (segmentIndex === 2) {
+                        const earlyWithdrawResult = await goodGhosting.earlyWithdraw({ from: secondLoser});
+                        truffleAssert.eventEmitted(
+                            earlyWithdrawResult,
+                            "EarlyWithdrawal",
+                            (ev) => ev.player === secondLoser,
+                            "loser unable to early withdraw from game",
+                        );
```

Comment:
In this scenario, does only player 2 performs an early withdraw? Should player 1 do the same or is it irrelevant?

Reply:
player 1 already does it above

Reply:
Got it. I miss it above. Thanks for pointing it out.